vllm_spyre/platform.py

```diff
@@ -540,3 +513,72 @@ def get_max_output_tokens(self, prompt_len: int) -> int:
                 max_new_tokens = max(max_new_tokens, shape['new_tokens'])
 
         return max_new_tokens
+
+    @classmethod
+    def configure_granite_3_8b(cls, vllm_config: VllmConfig):
+        """
+        Configure hard coded values for the model
+        https://huggingface.co/ibm-granite/granite-3.3-8b-instruct
+        """
+        parallel_config = vllm_config.parallel_config
+
+        if parallel_config.world_size != 4:
+            # only override configs for TP=4
+            return
+
+        tkv_128k = 128 * 1024
+        if not os.getenv("VLLM_DT_MAX_BATCH_TKV_LIMIT", None):
+            os.environ["VLLM_DT_MAX_BATCH_TKV_LIMIT"] = str(tkv_128k)
+            logger.info("Model granite-3.3-8b-instruct and tensor parallel " \
+            "size 4 detected. Using VLLM_DT_MAX_BATCH_TKV_LIMIT = %d",
+            tkv_128k)
+        else:
```

Comment:
NIT: This `else` doesn't check the value of the env var, so a warning will be produced even if set to the expected value.

Could make it an
```
elif os.getenv("VLLM_DT_MAX_BATCH_TKV_LIMIT") != str(tkv_128k):
```